packages/utils/pack-up/src/node/core/git.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Extracted & reduced from https://github.com/jonschlinkert/parse-git-config/blob/master/index.js
+ */
+import fs from 'fs/promises';
+import ini from 'ini';
+import os from 'os';
+import path from 'path';
+
+const resolveConfigPath = async ({ cwd }: { cwd: string }) => {
+  const configPath = path.join(os.homedir(), '.gitconfig');
+
+  try {
+    await fs.access(configPath);
+    return path.resolve(cwd, configPath);
+  } catch (err) {
+    return null;
+  }
+};
+
+interface GitConfig {
```

Comment:
I think I get what is going on here (typing the .gitconfig file?) but can you add a comment to this file explaining what's going on so nobody has to think about it? :)